test/e2e/tests/test_version.py

```diff
@@ -116,14 +118,56 @@ def test_smoke(self, lambda_client, lambda_function):
         lambda_validator = LambdaValidator(lambda_client)
 
         version_number = cr['status']['version']
-
+        assert version_number == "1"
+         
         # Check version exists
         assert lambda_validator.version_exists(lambda_function_name, version_number)
 
+        # Updating Function code
+        fn_resource["spec"]["description"] = "Updated descriptionssss"
+
+        # Patch k8s resource for Function
+        k8s.patch_custom_resource(function_reference,fn_resource)
+        time.sleep(UPDATE_WAIT_AFTER_SECONDS)
+        
+        # Publishing new version
+        resource_name_v2 = random_suffix_name("lambda-version", 24)
+        replacements["VERSION_NAME"] = resource_name_v2
+
+        resource_data_v2 = load_lambda_resource(
+            "version",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Creating new Version resource
+        ref_v2 = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name_v2, namespace="default",
+        )
+        k8s.create_custom_resource(ref_v2, resource_data_v2)
+        cr_v2 = k8s.wait_resource_consumed_by_controller(ref_v2)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref_v2)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        cr_v2 = k8s.wait_resource_consumed_by_controller(ref_v2)
+
+        lambda_validator = LambdaValidator(lambda_client)
+
+        version_number = cr_v2['status']['version']
+
+        assert version_number == "2"
+
```

Comment:
For completeness, you would need another
```
assert lambda_validator.version_exists(lambda_function_name, version_number)
```
over here before deleting it

Reply:
Added, thanks!